### DISCUSSION TITLE
Restrict upload mime types

### DISCUSSION THREAD

**Discussion by SuddenDev** [payloadcms/payload#176] [Feature Requests & Ideas]
So I read again through the docs, but could not find any suggestion how to limit uploads to a certain mime type. I'm sure I could just use a hook and check the incoming data before saving and return an error, but for the user experience it's a lot nicer if the input on the client "blocks" certain filetypes (by hiding them) at least on the surface. You'd still need to do a check on the server though. 

Any suggestions on how to tackle something like this? Or do you guys think it's enough to just check once upoaded?
Cheers!

**Comment by jmikrut** [comment]
I think this would be a great feature request!

I too think it would be a nice experience to block certain mime types by "graying them out" right in the upload UX, as other apps do.

To get that to work, we'd need a config-based approach. It would be pretty easy though. That would also alleviate the need to write a custom hook or, better yet—validator on the `filename` field that checks the data.

We'll get that done no problem. Good call!

**Reply by SuddenDev** [reply]
Great to hear that! Looking forward to it!

**Comment by denolfe** [comment]
This feature has been released in 0.6.8! The configuration is used for both the file picker and validation on the server.
Documentation can be found here: https://payloadcms.com/docs/upload/overview#mimetypes

@SuddenDev

**Reply by SuddenDev** [reply]
Fantastic! Great job, thanks for the fast implementation! @denolfe @jmikrut

**Reply by TomSmedley** [reply]
@denolfe It would be great to see this option on the 'upload' field too. Let's say you limit to 'image/*' and 'video/*' but you have a field you just want the user to upload images, you can use validation but it will still upload then create an error taking up disk space. At the moment I've had to create two collections.
